src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -128,43 +122,36 @@ public Result charTyped(char charTyped,
 
     @Nullable
     private static LSPFormattingFeature getFormattingFeature(@NotNull PsiFile file) {
+        if (ProjectIndexingManager.isIndexingAll()) {
+            // while indexing, we do nothing
+            return null;
+        }
+        if (!hasLanguageServerSupportingOnlyFormatting(file)) {
+            // The file is not associated to a language server which supports only formatting
+            return null;
+        }
         Project project = file.getProject();
         VirtualFile virtualFile = file.getVirtualFile();
 
-        CompletableFuture<@NotNull List<LanguageServerItem>> languageServersFuture = LanguageServiceAccessor.getInstance(project).getLanguageServers(
-                virtualFile,
+        CompletableFuture<@NotNull List<LanguageServerItem>> future = LanguageServiceAccessor.getInstance(project)
```

Comment:
I don't know if it's really important with the addiitonal check you've done, but I think this future can be avoided altogether. I'll chat with you about it directly.